core/ir/synthetic/encoding_common.h

```diff
@@ -33,31 +33,36 @@
 #ifndef _SYNTHETIC_ENCODING_COMMON_H_
 #define _SYNTHETIC_ENCODING_COMMON_H_
 
-/* This synthetic ISA is a made up ISA that has the purpose of preserving register
+/**
+ * This synthetic ISA is a made up ISA that has the purpose of preserving register
  * dependencies and giving hints on the type of operations each instruction is performing.
  * For this reason the majority of operations that would normally work on instructions
  * coming from an actual ISA are not supported.
- * The only operations that we support are instr_encode() and decode() when DynamoRIO's
- * host is an x86 architecture.
- * Also note that in DynamoRIO we use the #instr_t ISA mode to determine what type of
- * encoding to perform, and #dr_context_t ISA mode to determine what type of decoding to
- * perform.
- * Currently the only exception to this rule happens for decoding of synthetic
- * instructions, where #instr_t ISA mode takes precedence and #dr_context_t ISA mode is
- * ignored if #instr_t ISA mode is DR_ISA_SYNTHETIC.
- * XXX i#1684: Note that this is part of a larger issue, where lack of cross-arch support
+ * The operations we currently support are instr_encode_to_copy() to encode an #instr_t
```

Comment:
instr_encode() would also work

Reply:
Yes, I didn't want to list all the functions that call `instr_encode_to_copy()` (or `decode_common()`), as they might change or we might add more (although, likely, not frequently) and having to keep this comment up to date outweights the benefit of having such complete info IMHO.
I can add a list, if we deem it necessary though.

Reply:
I think it will be confusing to users as is: how would a user know whether instr_encode() works?  A denylist may be the way to go: If all encoding operations work, maybe just say that, and say that all decoding operations work except decode_sizeof() (is that the only one?).

Reply:
Actually, a list of what works might be shorter.
Here's the list for decode-related operations (following the same order of `core/ir/decode_api.h`):
```
OPERATION | SUPPORTED? (YES/YES*with a caveat/NO) | A BIT OF CONTEXT (if needed)
decode_eflags_usage() NO (breaks)
decode() YES
decode_from_copy() YES
decode_as_bb() NO (prob break)
decode_trace() NO (prob break)
decode_first_opcode_byte() YES* (ISA doesn't matter, returns op_instr[] of whatever enabled ISA of DR build, hence not the "intended behavior" if a user has DR_ISA_REGDEPS in mind... there is no opcode there)
decode_opcode_name() YES* (ISA doesn't matter, similar to decode_first_opcode_byte(), not intended behavior)
set_x86_mode() YES* (ISA doesn't matter)
get_x86_mode() YES* (ISA doesn't matter)
dr_app_pc_as_jump_target() NO (doesn't break, worst case in ARM a bit of PC is flipped, but DR_ISA_REGDEPS doesn't need it?)
dr_app_pc_as_load_target() NO (doesn't break, worst case in ARM a bit of PC is flipped, but DR_ISA_REGDEPS doesn't need it? <--- similar to dr_app_pc_as_jump_target())
decode_memory_reference_size() YES* (not tested though, seems like it would just return 0)
decode_sizeof() NO (prob break, but easy to implement for DR_ISA_REGDEPS)
decode_sizeof_ex() NO (prob break, but easy to implement)
decode_next_pc() NO (calls decode_sizeof(), prob break, but easy to implement)
decode_cti() NO (prob break)
```

And a list for encode-related operations (following the same order of `core/ir/encode_api.h`):
```
OPERATION | SUPPORTED? (YES/YES*with a caveat/NO) | A BIT OF CONTEXT (if needed)
dr_set_isa_mode() YES
dr_get_isa_mode() YES
dr_set_sve_vector_length() YES* (ISA doesn't matter, returns data based on sve_veclens)
dr_get_sve_vector_length() YES* (ISA doesn't matter, similar to dr_set_sve_vector_length())
instr_is_encoding_possible NO (prob break, uses instr_t_info, which DR_ISA_REGDEPS doesn't support, might be easy to implement)
instr_encode() YES
instr_encode_to_copy() YES
instrlist_encode() YES* (not tested, but seems to be calling supported operations of instr_t like instr_length(), instr_num_srcs(); then it just calls instr_encode_to_copy(), which is supported)
instrlist_encode_to_copy() YES* (not tested, same as instrlist_encode())
```

We should probably just list the YES.